### ISSUE TITLE
No use of activation function in decoder ?

### ISSUE THREAD

**Issue by gregjauvion** [y0ast/VAE-TensorFlow#7]
Hello,

I want to write a VAE with tensorflow, and I was looking at your implementation. I am surprised by this line : x_hat = tf.matmul(hidden_decoder, W_decoder_hidden_reconstruction) + b_decoder_hidden_reconstruction

Isn't it more usual to add an activation function to build the output from the decoder's hidden layer, i.e doing something like x_hat = **tf.nn.sigmoid**(tf.matmul(hidden_decoder, W_decoder_hidden_reconstruction) + b_decoder_hidden_reconstruction) ?

Thanks!
Grégoire

**Comment by gregjauvion** [comment]
Oh, it seems that the function tf.nn.sigmoid_cross_entropy_with_logits(logits=x_hat, labels=x) computes the cross-entropy between sigmoid(x_hat) and x actually, so your code is OK I think...
If my understanding is good, the reconstructed x is actually sigmoid(x_hat) and not x_hat...

**Comment by y0ast** [comment]
Yep, let me know if you have more questions.
